tests/unit/cli/vyper_compile/test_compile_files.py

```diff
@@ -364,6 +366,50 @@ def test_archive_output(input_files):
     out2 = compile_files([archive_path], ["integrity", "bytecode"])
     assert out[contract_file] == out2[archive_path]
 
+    # tamper the integrity sum by using the resolved imports hash, which excludes the storage layout
+    with (
+        library_file.open() as f,
+        contract_file.open() as g,
+        jsonabi_file.open() as h,
+        storage_layout_path.open() as i,
+    ):
+        library_contents = f.read()
+        contract_contents = g.read()
+        jsonabi_contents = h.read()
+        storage_layout_contents = i.read()
+
+    contract_hash = sha256sum(contract_contents)
+    library_hash = sha256sum(library_contents)
+    jsonabi_hash = sha256sum(jsonabi_contents)
+    resolved_imports_hash = sha256sum(contract_hash + sha256sum(library_hash) + jsonabi_hash)
+    storage_layout_hash = sha256sum(storage_layout_contents)
+    expected_hash = sha256sum(storage_layout_hash + resolved_imports_hash)
+
+    integrity_filename = "MANIFEST/integrity"
+    with zipfile.ZipFile(archive_path, "r") as zip_read:
+        memory_zip = io.BytesIO()
+
+        with zipfile.ZipFile(memory_zip, "w") as zip_write:
+            for item in zip_read.infolist():
+                if item.filename == integrity_filename:
+                    zip_write.writestr(item, resolved_imports_hash.encode())
+                else:
+                    zip_write.writestr(item, zip_read.read(item.filename))
+
+        memory_zip.seek(0)
+        with open(archive_path, "wb") as f:
+            f.write(memory_zip.read())
```

Comment:
the seek looks a bit weird
```suggestion
        with open(archive_path, "wb") as f:
            f.write(bytes(memory_zip.buffer()))
```